src/drivers/cmakeDriver.ts

```diff
@@ -868,7 +868,23 @@ export abstract class CMakeDriver implements vscode.Disposable {
 
     private configRunning: boolean = false;
 
-    private buildRunning: boolean = false;
+    private _buildRunning: boolean = false;
```

Comment:
Something I don't like about get/set properties that wrap another piece of data is that there's no protection against this code accidentally setting `_buildRunning` directly.  Consider wrapping this functionality in a small class that manages this state and the "requested targets".

Reply:
I reverted `buildRunning` to what it was before.
I added new `CMakeBulldTaskRunner` class but I haven't moved `buildRunning` to it. Please let me know if you expect it there.

Reply:
Have just done final refactor `CMakeBuildTaskRunner' is renamed to `CMakeBuildRunner` and tracks also Subprocess of non-task based builds.
buildRunning flag was moved into this class.